### PR TITLE
Update README regarding changes in jemalloc

### PR DESCRIPTION
**Pull request by michalbiesek**
https://github.com/memkind/memkind/pull/351#issuecomment-619910583
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/353)

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -117,8 +117,10 @@ for more information about either the memkind or the jemalloc configuration opti
 ## jemalloc
 The jemalloc source was forked from jemalloc version 5.2.1.  This source tree
 is located within the jemalloc subdirectory of the memkind source.  The jemalloc
-source code has been kept close to the original form, and in particular
-the build system has been lightly modified.
+source code has been kept close to the original form, with exceptions:
```
Comment (jschmieg) [inline README:120]: source code has been kept close to the original form, except for the following items:

File: README [hunk 2]
```diff
@@ -117,8 +117,10 @@ for more information about either the memkind or the jemalloc configuration opti
...
+- extend jemalloc API with "arenalookupx" function
+- optimization for looking correct command in mallctl
```
Comment (jschmieg) [inline README:122]: optimization for searching commands in mallctl

File: README (hunk 1 above)
Comment (michalbiesek) [reply README]: Done.

File: README (hunk 2 above)
Comment (michalbiesek) [reply README]: Done.

**Pull request merged**